### PR TITLE
Move README to Go doc

### PR DESCRIPTION
**Pull request by junr03**
Move parts of the README to Go doc. 

@rodaine

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,7 @@
 ## Overview
```
Comment (rodaine) [inline README.md]: Is the TOC still needed?

File: README.md [hunk 2]
```diff
@@ -39,143 +40,9 @@ make bootstrap && make compile-test
...
-Timers are added to a store, or a scope by using the `NewTimer(name string)` method.
-
-Timespans can be used to measure spans of time. They measure time from the time they are allocated by a Timer with `AllocateSpan()`,
-until they call `Complete()`. When `Complete()` is called the timespan is flushed. A Timespan can be flushed at function
-return by calling `Complete()` with golang's `defer` statement.
-
-### Stats with Tags
-
-Counters, Gauges and Timers can have tags in addition to their names. All three have methods 
-`New[Counter|Gauge|Timer]WithTags(name string, tags map[string]string)`.
```
Comment (rodaine) [inline README.md]: I'd recommend an example usage of some of these stats in the readme (as well as the godocs, which can be a different PR when you have the opportunity).

File: mock_sink.go [hunk 3]
```diff
@@ -6,6 +6,7 @@ type mockSink struct {
 	Gauges   map[string]uint64
 }
 
+// Returns a Mock Sink that flushes stats to in memory maps.
```
Comment (rodaine) [inline mock_sink.go]: golint will complain here. All comments should begin with the symbol name or an article (`the`, `a`, `an`). So this could be rephrased as `// NewMockSink returns a Sink that flushes to an in-memory map`

File: package.go [hunk 4]
```diff
@@ -0,0 +1,3 @@
+// Gostats is a Golang statistics library created by Engineers at Lyft
```
Comment (rodaine) [inline package.go]: package level comments should start with a short label and match the actual package name (not the repo/directory name): `// The stat package provides ...`

File: README.md (hunk 1 above)
Comment (junr03) [inline README.md]: removed

File: README.md (hunk 2 above)
Comment (junr03) [inline README.md]: 👍

File: mock_sink.go (hunk 3 above)
Comment (junr03) [inline mock_sink.go]: I ran golint to fix all the outstanding issues

File: package.go (hunk 4 above)
Comment (junr03) [inline package.go]: fixed

File: mock_sink.go (hunk 3 above)
Comment (rodaine) [inline mock_sink.go]: Sounds good!
